package.json

```diff
@@ -9,7 +9,7 @@
     "sleepDays": 7,
     "semverRange": "<2.0.0"
   },
-  "main": "lib/start.js",
+  "main": "generated/etcher.js",
```

Comment:
would you elaborate on how the entry point works atm?

I'm a bit confused by the fact, we remove usage of lib/start here, but keep using it in webpack config and update it to use `gui/etcher` instead of `generated/etcher`.

Reply:
Etcher has 2 "main" functions:
 * one launches the GUI: `lib/gui/etcher.js`;
 * the other launches the child writer process (when the env var `ELECTRON_RUN_AS_NODE` is set): `lib/gui/modules/child-writer.js`.

`lib/start.js` is the entrypoint that decides which one of the above needs to run (based on `ELECTRON_RUN_AS_NODE` being set or not)

Before this PR, the entrypoint in `package.json` was `lib/start.js` and it started either `generated/etcher.js` or `lib/gui/modules/child-writer.js` which is weird because one is webpacked and the other not). This means the same code may be ran from different files.

After this PR, the only entrypoint is `generated/etcher.js`, `lib/start.js` and all the things it requires (except node_modules) are webpacked.


In the long term, we shouldn't use webpack at all for the main process (we should only use it for the js running in the renderer process). But for now it is useful as we have mixed js & ts code (and also some [non standard stuff](https://github.com/balena-io/etcher/blob/master/webpack.config.js#L43)).

Reply:
Got it, thanks for the explanation.